src/main/java/uk/gov/hmcts/ccd/data/casedetails/DefaultCaseDetailsRepository.java

```diff
@@ -157,6 +174,27 @@ public PaginatedSearchMetadata getPaginatedSearchMetadata(MetaData metaData, Map
         return sr;
     }
 
+    // TODO This accepts null values for backward compatibility. Once deprecated methods are removed, parameters should
+    // be annotated with @NotNull
+    private Optional<CaseDetailsEntity> find(String jurisdiction, Long id, String reference) {
+        final CaseDetailsQueryBuilder qb = queryBuilderFactory.select(em);
+
+        if (null != jurisdiction) {
+            qb.whereJurisdiction(jurisdiction);
+        }
+
+        if (null != reference) {
+            qb.whereReference(reference);
+        } else {
+            qb.whereId(id);
+        }
+
+        final TypedQuery<CaseDetailsEntity> query = qb.build();
+        return query.getResultList()
```

Comment:
can we move this into a getSingleResult method?

Reply:
`getSingleResult()` would throw an exception on empty result we would then need to be caught to return an empty optional. I think `.getResultList().stream().findFirst()` gives a more fluent approach than exception handling.

Reply:
sorry didn't mean using getSingleResult of jpa. I meant create  a method on the builder and move this whole code in there

Reply:
oh sorry, I misunderstood. Can do, yes, good idea.

Reply:
Done.